.github/workflows/build-and-run-tests-windows.yml

```diff
@@ -0,0 +1,50 @@
+name: Build and run tests on Windows
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        required: true
+        type: string
+
+jobs:
+  build-and-run-tests-windows:
+    runs-on: ${{ inputs.platform }}
+    steps:
+    - run: echo "Job triggered by ${{ github.event_name }} event, running on ${{ runner.os }}."
+    - run: echo "Branch ${{ github.ref }} on repository ${{ github.repository }}"
+
+    - name: Check out repository code
+      uses: actions/checkout@v3
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Install Python Dependencies
+      run: pip install meson ninja
+
+    - name: Prepare MSVC
+      uses: bus1/cabuild/action/msdevshell@v1
```

Comment:
This in fact duplicates our `make_vs`. We should either drop that helper script (which I wont), or use our script to check if that still works.

Reply:
I think the `make_vs test` does only run the tests and no asan etc

yes...
```
%MESON% --buildtype=%BUILDTYPE% "%REPO_ROOT%%FOLDER%" "%REPO_ROOT:~0,-1%"
ninja -C "%REPO_ROOT%%FOLDER%" %TARGET%
```

Do we need an asan build on windows? Do we expect it finds address issues we do not find on other platforms?

The current state of this PR has two concurrent windows ci runs:
`windows.yml` just runs the tests, is independent of `ci.yml`, triggers only when it is needed
`build-and-run-tests-windows.yml` is called from `ci.yml` (which triggers on any push) and runs asan.

Of course one can combine somehow the two workflows, pick what is best of each.

>  But if you have a better solution, go ahead

Well, I already put commits in the PR. There just needs to be a decision what we want. The code is there and we can mix.

Further I find the names are too long. Both of the files themselves as well as the workflow names. It even gets gut off:

![image](https://github.com/gul-cpp/gul14/assets/16012374/ebe2adae-f705-483b-b922-8266cfa35fd3)

![image](https://github.com/gul-cpp/gul14/assets/16012374/c475d105-e336-41f1-89da-54739b1c10ab)

Reply:
> Do we need an asan build on windows? Do we expect it finds address issues we do not find on other platforms?

I would argue we don't exactly *need* any CI at all, but every different platform/sanitizer/compiler helps a bit. So maybe we should support ASAN from our batch file at some point. Maybe not in this PR.

Reply:
> Of course one can combine somehow the two workflows, pick what is best of each.

`windows.yml` works for me, but then we need an equivalent rewrite of `ci.yml` and the Posix file.